ComponentKit/StatefulViews/CKStatefulViewReusePool.mm

```diff
@@ -94,8 +100,24 @@ - (void)enqueueStatefulView:(UIView *)view
   NSAssert([NSThread isMainThread], nil);
   NSParameterAssert(view != nil);
   NSParameterAssert(controllerClass != nil);
+  
+  // maximumPoolSize will be -1 by default
+  NSInteger maximumPoolSize = [controllerClass maximumPoolSize];
+  
+  if (maximumPoolSize < 0 || [self _sizeForControllerClass:controllerClass context:context] < maximumPoolSize) {
+    // only add the stateful view to the pool if a maximum limit has not been set
+    return _pool[std::make_pair(controllerClass, context)].addView(view);
+  }
+}
 
-  return _pool[std::make_pair(controllerClass, context)].addView(view);
+- (NSInteger)_sizeForControllerClass:(Class)controllerClass
+                             context:(id)context
+{
+  const auto it = _pool.find(std::make_pair(controllerClass, context));
```

Comment:
This is a little wasteful because we're making the pair and looking it up in the pool twice.

Better might be to get rid of this helper method entirely and take an iterator into the pool like so:

```
auto it = _pool[std::make_pair(controllerClass, context)];
if (maximumPoolSize < 0 || it->second.viewCount() < maximumPoolSize) {
  it->second.addView(view);
}
```

But, it's your call.